api/src/main/java/org/pmiops/workbench/cdrselector/WorkspaceResourcesServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+      ConceptSetService conceptSetService,
+      DataSetDao dataSetDao,
+      WorkspaceMapper workspaceMapper) {
+    this.cohortReviewService = cohortReviewService;
+    this.conceptSetService = conceptSetService;
+    this.dataSetDao = dataSetDao;
+    this.workspaceMapper = workspaceMapper;
+  }
+
+  @Override
+  public List<WorkspaceResource> getWorkspaceResources(
+      DbWorkspace dbWorkspace, WorkspaceAccessLevel workspaceAccessLevel, List<ResourceType> resourceTypes) {
+    if (resourceTypes.size() == 0) {
+      throw new BadRequestException("Must provide at least one resource type");
+    }
+
+    List<WorkspaceResource> workspaceResources = new ArrayList<>();
+
+    if (resourceTypes.contains(ResourceType.COHORT)) {
+      final Set<DbCohort> cohorts = dbWorkspace.getCohorts();
+      workspaceResources.addAll(
+          cohorts.stream()
+              .map(
+                  cohort ->
+                      workspaceMapper.dbWorkspaceAndDbCohortToWorkspaceResource(
+                          dbWorkspace, workspaceAccessLevel, cohort))
+              .collect(Collectors.toList()));
+    }
+    if (resourceTypes.contains(ResourceType.COHORT_REVIEW)) {
+      List<DbCohortReview> reviews =
+          cohortReviewService.getRequiredWithCohortReviews(
+              dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName());
+      workspaceResources.addAll(
+          reviews.stream()
+              .map(
+                  cohortReview ->
+                      workspaceMapper.dbWorkspaceAndDbCohortReviewToWorkspaceResource(
+                          dbWorkspace, workspaceAccessLevel, cohortReview))
+              .collect(Collectors.toList()));
+    }
+    if (resourceTypes.contains(ResourceType.CONCEPT_SET)) {
+      List<DbConceptSet> conceptSets =
+          conceptSetService.findByWorkspaceId(dbWorkspace.getWorkspaceId());
+      workspaceResources.addAll(
+          conceptSets.stream()
+              .map(
+                  dbConceptSet ->
+                      workspaceMapper.dbWorkspaceAndDbConceptSetToWorkspaceResource(
+                          dbWorkspace, workspaceAccessLevel, dbConceptSet))
+              .collect(Collectors.toList()));
+    }
+    if (resourceTypes.contains(ResourceType.DATASET)) {
+      List<DbDataset> datasets =
+          dataSetDao.findByWorkspaceIdAndInvalid(dbWorkspace.getWorkspaceId(), false);
+      workspaceResources.addAll(
+          datasets.stream()
+              .map(
+                  dbDataset ->
+                      workspaceMapper.dbWorkspaceAndDbDatasetToWorkspaceResource(
+                          dbWorkspace, workspaceAccessLevel, dbDataset))
+              .collect(Collectors.toList()));
+    }
+    if (resourceTypes.contains(ResourceType.COHORT_SEARCH_GROUP) || resourceTypes.contains(ResourceType.COHORT_SEARCH_ITEM) || resourceTypes.contains((ResourceType.NOTEBOOK)) || resourceTypes.contains((ResourceType.WORKSPACE))) {
```

Comment:
This statement is brittle, since we could add other resource types later and they might not get added here. If you really want to validate these things, I'd do a SUPPORTED_RESOURCE_TYPES.containsAll(resourceTypes) at the top.

But what might be the bigger flag here is that this endpoint only supports a subset of the enum values in `ResourceType`. I think it wold be simpler if you had a `WorkspaceResourceType` and supported all of those. Then you'd never be able to  have this error (by definition).

Reply:
I think we will want to add the other resource types eventually. (I don't know some of them, @freemabd1 might know better) 

But I will move to the SUPPORTED_RESOURCE_TYPES model. I think a list of supported types is probably better than a list of unsupported types.